common/myservice/MySerializer.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace common\myservice;
+
+use yii\base\Component;
+use yii\helpers\Json;
+
+class MySerializer extends Component
+{
+    public $data;
+    public $pathUploadFile;
+
+    public function init()
+    {
+        parent::init();
+    }
+
+    /**
+     * @param $data
+     * @return string
+     */
+    public function Json($data)
+    {
+        return $data = Json::encode($data);
+    }
+
+    /**
+     * @param $data
+     */
+    public function save($data)
```

Comment:
This method should not be here. It is the same for both json and xml encoders. Please move it to writer service or some other service.